src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -584,10 +569,12 @@ private struct CompilationInfo
                 public Compilation Compilation { get; }
                 public bool HasCompleteReferences { get; }
 
-                public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
+                public CompilationInfo(Compilation compilation, ProjectInfo projectInfo, bool hasCompleteReferences)
                 {
                     this.Compilation = compilation;
-                    this.HasCompleteReferences = hasCompleteReferences;
+
+                    // to have complete reference, projectInfo should be complete and we could create complete reference for this compilation.
+                    this.HasCompleteReferences = projectInfo.IsComplete && hasCompleteReferences;
```

Comment:
Should we clarify at the ProjectInfo point that this is a matter of having complete _references_ to distinguish from other complete-or-non-completeness?